PubHub/PubHub.API/Controllers/BooksController.cs

```diff
@@ -474,8 +490,10 @@ public async Task<IResult> UpdateBookAsync(Guid id, [FromBody] BookUpdateModel u
         [ProducesResponseType(StatusCodes.Status404NotFound, Type = typeof(ProblemDetails))]
         public async Task<IResult> DeleteBookAsync(Guid id, [FromHeader] string appId)
         {
-            if (!_whitelistService.TryVerifyApplicationAccess(appId, GetType().Name, out IResult? problem))
-                return problem;
+            if (!_accessService.AccessFor(User, appId)
+                .CheckWhitelistEndpoint(GetType().Name)
```

Comment:
This is an endpoint for deleting books. Should this not exclusivly be accessible for publishers?

Maybe chekc other deletion endpoints to ensure the poliy is cocrrectly setup up.

Reply:
I do another check a bit further down on line 511 after retrieving the book 😉

Reply:
Okay. Why not exclude users earlier? Why retrieve the book data and then check?

Reply:
Because the check is based on who owns the book. We only get to know that after retrieving it 😊